### PR TITLE
Styling improvements for progresslist, checkbox and buttons

### LINKED ISSUE #298: Missing focus state in form section navigation
The items in the section menu in data entry forms lack a focus state. 
This has been added to design. See [Figma](https://www.figma.com/design/zZlFr8tYiRyp4I26sh6eqp/Kiesraad---Abacus-optelsoftware?node-id=123-2660&node-type=FRAME&t=lzcBvjqXEZ7UVHM0-11).

### Acceptance criteria
- [x] Focus can only be set on items that the user can navigate to (so it should not be possible to tab to items that are disabled or active)
- [x] When an item has both focus and hover, the focus state overrides the hover

image

### PR DESCRIPTION
**Pull request by Lionqueen94**
Closes #298

### CODE REVIEW

**Review by jorisleker — CHANGES_REQUESTED** [review]
See https://github.com/kiesraad/abacus/issues/298#issuecomment-2366436111

File: frontend/lib/ui/ProgressList/ProgressList.module.css [hunk 1]
```diff
@@ -10,28 +10,41 @@
     display: flex;
     align-items: center;
     gap: 0.5rem;
+    padding: 0.625rem 0 0.625rem 0.75rem;
 
-    &:hover {
+    :not(&:global(.ruler)):not(&:global(.disabled)):not(&:global(.active))&:hover {
```
Comment (lkleuver) [inline frontend/lib/ui/ProgressList/ProgressList.module.css:15]: i find this a bit hard to grok.
Imho it's nicer to have default styles and override them in variants.

File: frontend/lib/ui/ProgressList/ProgressList.module.css [hunk 2]
```diff
@@ -10,28 +10,41 @@
...
       background-color: var(--color-hover);
     }
 
+    :not(&:global(.ruler)):not(&:global(.disabled)):not(&:global(.active))&:focus {
```
Comment (lkleuver) [inline frontend/lib/ui/ProgressList/ProgressList.module.css:19]: idem

**Review by lkleuver — DISMISSED** [review]

File: frontend/lib/ui/ProgressList/ProgressList.module.css (hunk 1 above)
Comment (Lionqueen94) [reply frontend/lib/ui/ProgressList/ProgressList.module.css]: I have improved this

**Comment by jorisleker** [comment]
@Lionqueen94 reviewed again:
- focus state for buttons: 👌 
- focus state items in progress list: 👌 
- focus state checkbox: ❌ not working yet on macos (tested chrome, firefox and safari)

**Review by jorisleker — APPROVED** [review]
👌

**Review by lkleuver — DISMISSED** [review]

**Review by jorisleker — APPROVED** [review]

**Review by lkleuver — APPROVED** [review]

**Pull request merged**